project_task_schedule/models/project_project.py

```diff
@@ -0,0 +1,9 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    enable_task_planning = fields.Boolean("Enable Task Planning?")
```

Comment:
`date_planned_required` may be better, with a help text of 'When this is enabled, tasks on this project require a start and end date'